src/legacy/core_plugins/kibana/migrations/migrations.js

```diff
@@ -484,6 +484,31 @@ function migrateSubTypeAndParentFieldProperties(doc) {
   };
 }
 
+function migrateNumeralFieldFormatters(doc) {
+  if (!doc.attributes.fieldFormatMap) return doc;
+
+  const fieldFormatMap = doc.attributes.fieldFormatMap;
+  const fieldFormats = JSON.parse(fieldFormatMap);
```

Comment:
Is `fieldFormatMap` always going to be valid JSON? If even a single one is not, this will crash Kibana and prevent it from starting. We should probably handle this kind of error, here, if we can do something sane in the error case.